src/main.rs

```diff
@@ -241,15 +242,21 @@ ___  ___      _       _         _   _      _
 
     let mut passed_tests = 0;
     let mut failed_tests = 0;
+    let mut all_failed_tests: HashMap<String, String> = HashMap::new();
+
     println!("{}", ("Igniting tests 🔥\n").to_string().bright_red());
+
     test_suites.iter().for_each(|(key, val)| {
         println!("🧪 Running Test Suite: {}", key.blue());
         println!("{}\n", "=".repeat(50));
         logging::add_indent();
+
         for test in &val.tests {
-            if test.run().passed {
+            let result = test.run();
+            if result.passed {
                 passed_tests += 1;
             } else {
+                all_failed_tests.insert(test.name.clone(), result.logs);
                 failed_tests += 1;
             }
         }
```

Comment:
I was working on displaying the full path to the `test()` function (as in the screenshot), and one of the issues was that there might be some generic test names (e.g. "Should yield correct value") in different suites, so this might be the right way to do it. I will look into it today. Thanks!

<img width="819" alt="Screenshot 2021-12-02 at 9 33 23" src="https://user-images.githubusercontent.com/20456492/144377892-b01f0018-87b8-4414-8cdd-5866edf6689a.png">